### DISCUSSION TITLE
Temporary URLS for local filesystem

### DISCUSSION THREAD

**Discussion by helderneves91** [laravel/framework#52773] [Q&A]
Hi! I have updated the project to include the temporary url for local filesystems. I think I'm misunderstanding something, because I have get the error `Route [storage.exams] not defined.` when trying to build a url.

For context, the filesystem is:
```php
'exams' => [
    'driver' => 'local',
    'root' => storage_path('app/private/exams'),
    'url' => env('APP_URL').'/storage',
    'serve' => true,
    'throw' => false,
],
```

So, when I test this code, I got the error.
```php
$url = Storage::disk('exams')->temporaryUrl(
    $file, now()->addMinutes(1)
);
```

Any hints? What am I missing?

Thanks.

**Comment by newtonjob** [comment]
You've probably got your routes already cached. I think just clear the route cache and try again?

**Reply by helderneves91** [reply]
Hey, no that's not the case. I cleared the cache and the problem persists.

**Reply by newtonjob** [reply]
Just to be clear, in mean `route:clear` not `cache:clear`

**Reply by helderneves91** [reply]
Usually I do artisan optimize:clear, which includes routes.

**Comment by ChrisSantiago82** [comment]
Were you able to resolve this issue?
I'm running into the same issue with the local disk.
The documentation only says to set serve to true:

![image](https://github.com/user-attachments/assets/073c8718-9d4d-4ed9-8e9d-2ed1b00f38c1)

**Reply by helderneves91** [reply]
Hello, no, but I also didn't investigate further because the project was abandoned.

**Reply by ChrisSantiago82** [reply]
Thank you

**Reply by Quezler** [reply]
@ChrisSantiago82 i ran into it today too, looks like laravel throws a tantrum when there are 2 or more local disks **defined** with serve set to true, the local storage disks generate a route and the url defaults to storage for both, if you have multiple local disks give them all (or all but one) the url option set to an unique string, like storage-\<disk name>

**Reply by ChrisSantiago82** [reply]
Interesting, I thought maybe this is only an issue I'm having.
Maybe I can create a test an open an issue.

**Reply by Quezler** [reply]
did you by any chance set serve to true on the public disk as well, not just local @ChrisSantiago82?

**Comment by FlightlessParrot** [comment]
I had same issue. It seems that Laravel does not handle temporary links for custom disks. You can only set 
`'serve' => true `
in local dysk. Using this property  in any other disk gonna result in route.diskname undefined error. 

If you need temporary links using other disks you have to create the feature manually.

**Reply by ChrisSantiago82** [reply]
I had created an issue before, but was closed pretty fast.
The issue that I see is, you may use temporary links for a while in your app. Then you add a custom disk, unrelated to the temporary link, and your app breaks (somewhere else).
